test: improve language service tests performance

alan-agius4 opened the pull request:
With this change we reduce the amount of IO operations. This is especially a huge factor in windows since IO ops are slower.

With this change mainly we cache `existsSync` and `readFileSync` calls

Here's the results

Before
```
//packages/language-service/test:test
INFO: Elapsed time: 258.755s, Critical Path: 253.91s
```

After
```
//packages/language-service/test:test
INFO: Elapsed time: 66.403s, Critical Path: 63.13s
```

alan-agius4 commented:
On codefresh CI

Before
```
//packages/language-service/test:test                                    PASSED in 297.9s 
```

With the change
```
//packages/language-service/test:test                                    PASSED in 111.6s 
```

That shaves off approx 3mins

//cc @filipesilva

kyliau commented on packages/language-service/test/test_utils.ts:
```diff
@@ -20,8 +20,8 @@ export type MockDirectory = {
-const rxjsts = /rxjs\/(\w|\/)+\.tsx?$/;
-const rxjsmetadata = /rxjs\/(\w|\/)+\.metadata\.json?$/;
+const rxjsts = /rxjs\/(\w|\/|-)+\.tsx?$/;
+const rxjsmetadata = /rxjs\/(\w|\/|-)+\.metadata\.json?$/;
```
Please add an example in the comment for what the regexes are matching against?

alan-agius4 replied:
Actually removed this change

alan-agius4 committed:
test: improve language service tests performance

With this change we reduce the amount of IO operations. This is especially a huge factor in windows since IO ops are slower.

With this change mainly we cache `existsSync` and `readFileSync` calls

Here's the results

Before
```
//packages/language-service/test:test
INFO: Elapsed time: 258.755s, Critical Path: 253.91s
```

After
```
//packages/language-service/test:test
INFO: Elapsed time: 66.403s, Critical Path: 63.13s
```

IgorMinar commented:
Is it right that this is master only PR?

alan-agius4 commented:
@IgorMinar, no it’s should be master & patch.


Thanks for highlighting it.

matsko commented:
@alan-agius4 @IgorMinar I merged this only to master ... let me know if this was the correct thing to do.

alan-agius4 commented:
@matsko, can you please merge it to patch as well? Or it doesn’t apply cleanly?

matsko commented:
It doesn't apply cleanly. Can you create another PR based on the `patch` branch and I will merge it in?

matsko commented:
Patch being `8.x`

alan-agius4 commented:
@matsko, yes, will do that on Monday and I’ll ping you, as I away for the weekend.

alan-agius4 commented:
@matsko, here's the patch version https://github.com/angular/angular/pull/30683
